config/localTemplate.go

```diff
@@ -53,6 +53,14 @@ type Local struct {
 	GossipFanout int    `version[0]:"4"`
 	NetAddress   string `version[0]:""`
 
+	// wsNetwork.go
+	// if set, the node will use Identity based Connection Deduplication.
+	// Identity Challenges are initiated with an Address field which must match the intended recipient's gossip address
+	// example: ws://r-XX.algorand-mainnet.network:4160/v1/mainnet-v1.0/gossip
+	// received Identity Challenges which do not correctly specify this Address will be ignored.
+	// the peering will go forward without any further participation in identification.
+	ConnectionDeduplicationName string `version[27]:""`
```

Comment:
it is the entire gossip url, the `gossipAddr` in tryConnect

Reply:
(ws://.../v1/mainnet-v1.0/gossip)

Reply:
if this could _accept_ just "host:port" and do any normalization transformation later, that would be the most admin-user friendly

Reply:
not sure how you mean normalization, but I have changed it to `host:port` now.

Reply:
I found this `PublicAddress string 'version[0]:""'` in the config file, so maybe we can use this parameter instead?

Reply:
that would work, *except* we want to make this behavior opt-in, so if we started using `PublicAddress` for deduplication, it might surprise node operators  that the way we use the variable has changed. Making it `ConnectionDeduplicationName` allows for the operator to knowingly set the name

Reply:
the reuse concern is valid but 1) having two config vars with the same value does not look good. 2) I do not believe someone ever set `PublicAddress` in config.
Btw, the value of `ConnectionDeduplicationName` - can it be arbitrary name? Why can't we use `WebsocketNetwork.Address()` method instead and have a boolean variable?
How the code handles situations when two+ nodes set ConnectionDeduplicationName to the same value?

Reply:
:+1: Revision is now out using this value and dropping ConnectionDeduplicationName. I also added the comment from our configuration settings webpage, and expanded it to capture the conneciton dedup case.

Reply:
> Btw, the value of ConnectionDeduplicationName - can it be arbitrary name? 

It *could* be some arbitrary name, but if an operator does that, all the peers have to know to use that name somehow. And that's exactly why it's the PublicAddress, because the DNS name of the peer is the only thing we have that predefines names. If there was some separate authority that named all nodes, then we could use that.

> How the code handles situations when two+ nodes set ConnectionDeduplicationName to the same value?

Like cce mentioned, if multiple hosts share a deduplication name, the identity challenges will still pass and can be deduplicated (they could even connect to one another without issue, at least from the ID perspective). They would lack some protection because an actor could route an identity challenge meant for *one* to the *other*, but an attacker still couldn't use that to encircle a relay since *most* relays wouldn't share names.

> Why can't we use  WebsocketNetwork.Address() method instead and have a boolean variable? 

Load balancers and NAT mean that the DNS name a host might be using to reach out is not the internally known address.

@algorandskiy